docs/developers/architecture/app_model.md

```diff
@@ -2,66 +2,195 @@
 
 # napari's application model
 
-```{important}
-**This is not a part of the public napari interface!**
-
-This page is mostly aimed at developers who are interested in contributing to
-or understanding the inner workings of napari.
-```
-
 ```{warning}
 **Work in progress!**
-This document is here to give a little guidance on the current
-vision for how this application model might be used. It will very likely change
-as we develop it.
+The napari application model is currently being developed. This document is here to
+give guidance but things may change as we develop. It may link to current
+pull requests and issues which have more up to date information on specific
+areas.
 ```
 
 ## App-model
 
-The global application singleton can be retrieved with `napari._app_model.get_app()`.  It is an instance of
-[`app_model.Application`](https://app-model.readthedocs.io/en/latest/application/).
-`app-model` is a Python package that provides a declarative schema for an
-application.  It is an abstraction developed by napari developers, with the
+[`app-model`](https://app-model--142.org.readthedocs.build/en/142/) is a Python package
+that provides a declarative schema for a GUI-based
+application. It is an abstraction developed by napari developers, with the
 needs of napari in mind, but it is agnostic to napari itself (i.e. it should be
 reusable by any python GUI application).
 
-Currently, the primary purpose of the `app` is to compose the various
-[registries](https://app-model.readthedocs.io/en/latest/registries/) (commands,
-keybindings, menus, etc...) into a single name-spaced object.
+The {class}`~napari._app_model._app.NapariApplication` (`app`)
+is the top level object that stores information about the commands, keybindings
+and menus that make up the appliation.
+It is a subclass of {class}`app_model.Application` and is a global application
+singleton. It can be retrieved with {func}`napari._app_model.get_app`.
+
+Currently, the primary purpose of the `app` is to compose the following
+{mod}`app_model.registries` into a single name-spaced object:
+
+* {class}`~app_model.registries.CommandsRegistry`: maintains all of the
+  [commands](app-model-commands) (the actual callable objects) that have been
+  registered with the application. Accessible via `app.commands`.
+* {class}`~app_model.registries.MenusRegistry`: maintains all of the
+  [menu and submenus](appp-model-menus) that have been registered with the application.
+  Accessible via `app.menus`.
+* {class}`~app_model.registries.KeyBindingsRegistry`: maintains the association
+  between a [KeyBinding](app-model-keybindings) and a command id in the
+  {class}`~app_model.registries.CommandsRegistry`. Accessible via `app.keybindings`.
+
+The app-model
```

Comment:
```suggestion
The `app-model`
```

Reply:
Yeah I was not sure on this one. I think @jni said no backticks for napari, and I guess I thought it would be weird if napari package didn't get backticks but app-model did...? WDYT?